src/utilities/convex_hull.zig

```diff
@@ -185,10 +187,56 @@ pub const ConvexHull = struct {
         return LinearFunction{ .slope = min_slope, .intercept = min_intercept };
     }
 
+    /// Merges another `other: ConvexHull` into the current `self: ConvexHull` in place. This
+    /// operation modifies the `self` object by appending points from the `other` convex hull.
+    /// The function assumes both convex hulls are valid and sorted. If appending to the lower
+    /// or upper hull fails due to allocation issues, the error is propagated.
+    pub fn merge(self: *ConvexHull, other: *ConvexHull) !void {
+        // Special case: if `other` has only one point, directly add it to self's hulls
+        if (other.upper_hull.items.len == 1 and other.lower_hull.items.len == 1) {
+            try self.addPoint(other.upper_hull.items[0]);
+            return;
+        }
+
+        // Special case: if `self` has only one point, transfer points from `other` directly
+        if (self.upper_hull.items.len == 1 and self.lower_hull.items.len == 1) {
```

Comment:
Does `self` not contain two points if this if expression is `true`? This seems to contradict the comment.

Reply:
See https://github.com/cmcuza/TerseTS/pull/57#discussion_r1864272784